### PR TITLE
re-write mcafee atd in python

### PR DESCRIPTION
**Pull request by bakatzir**
fixes: https://github.com/demisto/etc/issues/12675

### CODE REVIEW

File: Integrations/integration-McAfee-ATD.yml [hunk 1]
```diff
@@ -36,687 +37,665 @@ configuration:
...
+        if demisto.params()['proxy']:
+            proxy['http'] = os.environ['http_proxy']
+            proxy['https'] = os.environ['https_proxy']
+        return proxy
+
+    ''' GLOBALS '''
+    USERNAME = demisto.params()['username']
+    PASSWORD = demisto.params()['password']
+    USE_SSL = not demisto.params()['unsecure']
+    BASE = load_server_url()
```
Comment (anara123) [inline Integrations/integration-McAfee-ATD.yml]: BASE -> BASE_URL

File: Integrations/integration-McAfee-ATD.yml [hunk 2]
```diff
@@ -1209,7 +1186,7 @@ script:
     - contextPath: IP.Address
       description: IP's relevant to the sample
       type: string
-    description: Deprecated, please use the 'McAfee ATD Detonate URL' playbook instead
+    description: Deprecated, please use detonate playbook instead.
```
Comment (anara123) [inline Integrations/integration-McAfee-ATD.yml]: write the playbook name

File: Integrations/integration-McAfee-ATD.yml [hunk 3]
```diff
@@ -36,687 +37,665 @@ configuration:
...
-    }
-  type: javascript
+        #elif demisto.command() == 'detonate-file-remote':
+            #return detonate(3, args.url, args.timeout, args.format, args.threshold);
+            #submit type for url-download is 3
+
+    except Exception, e:
+        LOG(e.message)
+        LOG.print_log()
+        raise
```
Comment (anara123) [inline Integrations/integration-McAfee-ATD.yml]: use return_error instead of raise

File: Integrations/integration-McAfee-ATD.yml [hunk 4]
```diff
@@ -36,687 +37,665 @@ configuration:
...
+        elif demisto.command() == 'atd-list-analyzer-profiles':
+            listProfiles()
 
-    function jobOrTaskId(){
-        uriSuffix = '';
-        if ( (!args.jobId && !args.taskId) || (args.jobId && args.taskId) ){
-            throw 'You must specify one (and only one) of the following: jobId, taskId.';
-        }
+        elif demisto.command() == 'atd-list-user':
+            listUsersCommand()
```
Comment (anara123) [inline Integrations/integration-McAfee-ATD.yml]: listUsersCommand => list_users_command (python conventions)

File: Integrations/integration-McAfee-ATD.yml [hunk 5]
```diff
@@ -36,687 +37,665 @@ configuration:
...
+            files=files
+        )
+        if res.status_code < 200 or res.status_code >= 300:
+                return_error('Got status code ' + str(res.status_code) + ' with body ' + res.content + ' with headers ' + str(res.headers))
+        uriStr = str(uri)
+        if (uriStr.find("session") == -1):
+            logout()
+        return res
+
+    def sendRequest(uri, method, headers = {}, body = {}, params={}, files={}):
```
Comment (anara123) [inline Integrations/integration-McAfee-ATD.yml]: call it `http_request` and unite sendRequest and sendRequestRaw

File: Integrations/integration-McAfee-ATD.yml [hunk 6]
```diff
@@ -36,687 +37,665 @@ configuration:
...
+            return int(val)
+        else:
+            return val
+
+    def getSessionCredentials():
+        res = sendRequest('php/session.php','get', LOGINHEADERS)
+        if (res):
+            return res['results']
+
+    def getHeaders():
```
Comment (anara123) [inline Integrations/integration-McAfee-ATD.yml]: create the headers in the beginning and then use them

File: Integrations/integration-McAfee-ATD.yml (hunk 1 above)
Comment (bakatzir) [reply Integrations/integration-McAfee-ATD.yml]: done

File: Integrations/integration-McAfee-ATD.yml (hunk 2 above)
Comment (bakatzir) [reply Integrations/integration-McAfee-ATD.yml]: done

File: Integrations/integration-McAfee-ATD.yml (hunk 3 above)
Comment (bakatzir) [reply Integrations/integration-McAfee-ATD.yml]: done

File: Integrations/integration-McAfee-ATD.yml (hunk 4 above)
Comment (bakatzir) [reply Integrations/integration-McAfee-ATD.yml]: done for all commands

File: Integrations/integration-McAfee-ATD.yml (hunk 6 above)
Comment (bakatzir) [reply Integrations/integration-McAfee-ATD.yml]: done

File: Integrations/integration-McAfee-ATD.yml (hunk 5 above)
Comment (bakatzir) [reply Integrations/integration-McAfee-ATD.yml]: done

**Pull request merged**